src/Core/Prefab/GlobeView.js

```diff
@@ -274,19 +274,23 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
 GlobeView.prototype = Object.create(View.prototype);
 GlobeView.prototype.constructor = GlobeView;
 
-GlobeView.prototype.addLayer = function addLayer(layer) {
-    if (layer.type == 'color') {
-        const colorLayerCount = this.getLayers(l => l.type === 'color').length;
-        layer.sequence = colorLayerCount;
-        layer.update = updateLayeredMaterialNodeImagery;
-    } else if (layer.type == 'elevation') {
-        if (layer.protocol === 'wmts' && layer.options.tileMatrixSet !== 'WGS84G') {
-            throw new Error('Only WGS84G tileMatrixSet is currently supported for WMTS elevation layers');
+GlobeView.prototype.addLayer = function addLayer(layer, parent) {
+    const duplicate = this.getLayers((l => l.id == layer.id));
+    if (duplicate.length === 0) {
```

Comment:
I don't think so.
If a user does this:
```js
view.addLayer({id: 'my_layer', ....});
view.addLayer({id: 'my_layer', ....});
```
Then the second layer needs preprocessing (or an error depending on the point of view);

A preprocessed layer can be identified because:
- it's a `Layer` (and not an object)
- it has a `layer.whenReady` property